app/controllers/teachers/sessions_controller.rb

```diff
@@ -1,33 +1,48 @@
 # frozen_string_literal: true
-
 class Teachers::SessionsController < Devise::SessionsController
   include TeacherCurrentNamespace
 
   layout "two_thirds"
 
   def create
-    if resource_params[:create_or_sign_in] == "create"
+    @new_session_form =
+      TeacherInterface::NewSessionForm.new(
+        email: new_session_form_params[:email],
+        create_or_sign_in: new_session_form_params[:create_or_sign_in]
+      )
+
+    if @new_session_form.create?
       redirect_to :eligibility_interface_countries
       return
     end
 
-    self.resource = resource_class.find_by(email: resource_params[:email])
+    if @new_session_form.valid?
+      self.resource = resource_class.find_by(email: @new_session_form.email)
 
-    if resource
-      if resource.active_for_authentication?
-        resource.send_magic_link
-      else
-        resource.resend_confirmation_instructions
+      if resource
+        if resource.active_for_authentication?
+          resource.send_magic_link
+        else
+          resource.resend_confirmation_instructions
+        end
+        redirect_to :teacher_check_email
       end
-
-      redirect_to :teacher_check_email
     else
-      set_flash_message(:alert, :not_found_in_database, now: true)
-      self.resource = resource_class.new(email: resource_params[:email])
-      render :new
+      render :new, status: :unprocessable_entity
     end
   end
 
+  def new
+    @new_session_form = TeacherInterface::NewSessionForm.new
+  end
```

Comment:
I'd suggest moving this function so it's defined above `create`, it's not a big deal but it [matches the general Rails convention](https://guides.rubyonrails.org/routing.html#crud-verbs-and-actions) so might make it faster to find the method for a new dev.